### DISCUSSION TITLE
I'm making a basic GUI for this

### DISCUSSION THREAD

**Discussion by Satanarious** [ALTaleX531/TranslucentFlyouts#29] [Show and tell]
I am making a basic GUI for this in Python. I could use some help and advice regarding the same as I have a few doubts. I was wondering if you could reach me out from links in my profile so we can discuss a bit. Or if you prefer, we can discuss it here as well.

Here is a sneek into the very bare-bone UI:
![image](https://github.com/ALTaleX531/TranslucentFlyouts/assets/42642682/05730ffb-723e-4aa0-bf7a-00c704347f0b)

**Comment by ALTaleX531** [comment]
Hello, Satanarious! Since it's not convenient for me to use Gmail, it's better to communicate with me directly on Github, and I was pleasantly surprised to hear that you made a GUI for this little project! After a brief look at the image, I'm curious about whether the settings in "General" are duplicated, and if so, maybe the image below is better to avoid unnecessary duplication. (edited by MSPaint)
![image](https://github.com/ALTaleX531/TranslucentFlyouts/assets/105535231/7c287a34-fb31-40ef-8172-437958758ccb)
In addition, I feel like using RGBA Sliders will be a little more user-friendly, and I'm looking forward to your future progress

**Reply by Satanarious** [reply]
Yes, after reading the config file, I figured out that the settings in 'General' are the same as what would be used in the 'DropDown' and 'Menu' sections. If I am not wrong then I think you are trying to imply that the 'Category' setting in the image you edited to avoid duplication in the 'DropDown' and 'Menu' sections should be used instead of the same settings in all 3 sections, when the value of the category is set to 'Global'. This would mean that Only the 'General' Section is displayed when the 'Category' setting is set to 'Global' and the other 2 sections should only be displayed when this setting is set to 'Individual', and the 'General' settings be disabled. I would share more sections of the UI after your response and once everything is confirmed, I will share my repository for you to create issues to further aid the development.

Edit: About the RGBA sliders, I'll make sure there's a colour picker for the same.

**Comment by Satanarious** [comment]
Please look at this JSON on how the structure of the GUI will look, its called defaults.json

```json
{
  "General": {
    "Scope": { "default": 0, "unsupported": 0 },
    "Effect Type": { "default": 5, "unsupported": 0 },
    "Enable Drop Shadow": { "default": 0, "unsupported": 0 },
    "Dark Mode Gradient Color": { "default": "", "unsupported": 0 },
    "Light Mode Gradient Color": { "default": "", "unsupported": 0 },
    "Dark Mode Opacity": { "default": "", "unsupported": 0 },
    "Light Mode Opacity": { "default": "", "unsupported": 0 },
    "Disabled": { "default": 0, "unsupported": 0 }
  },
  "DropDown": {
    "Effect Type": { "default": 5, "unsupported": 0 },
    "Enable Drop Shadow": { "default": 0, "unsupported": 0 },
    "Dark Mode Gradient Color": { "default": "", "unsupported": 0 },
    "Light Mode Gradient Color": { "default": "", "unsupported": 0 },
    "Dark Mode Opacity": { "default": "", "unsupported": 0 },
    "Light Mode Opacity": { "default": "", "unsupported": 0 },
    "Disabled": { "default": 0, "unsupported": 0 }
  },
  "Menu": {
    "No System Outline": { "default": 0, "unsupported": 0 },
    "Enable Immersive Style": { "default": 1, "unsupported": 11 },
    "Enable Custom Rendering": { "default": 0, "unsupported": 0 },
    "Effect Type": { "default": 5, "unsupported": 0 },
    "Enable Drop Shadow": { "default": 0, "unsupported": 0 },
    "Dark Mode Gradient Color": { "default": "", "unsupported": 0 },
    "Light Mode Gradient Color": { "default": "", "unsupported": 0 },
    "Dark Mode Opacity": { "default": "", "unsupported": 0 },
    "Light Mode Opacity": { "default": "", "unsupported": 0 },
    "Disabled": { "default": 0, "unsupported": 0 }
  },
  "Menu Items": {
    "Hot": {
      "Dark Mode Gradient Color": { "default": "", "unsupported": 0 },
      "Light Mode Gradient Color": { "default": "", "unsupported": 0 },
      "Dark Mode Opacity": { "default": "", "unsupported": 0 },
      "Light Mode Opacity": { "default": "", "unsupported": 0 },
      "Disabled": { "default": 0, "unsupported": 0 },
      "Corner Radius": { "default": 0, "unsupported": 0 },
      "Enable Theme Customization": { "default": 0, "unsupported": 0 }
    },
    "Disabled Hot": {
      "Dark Mode Gradient Color": { "default": "", "unsupported": 0 },
      "Light Mode Gradient Color": { "default": "", "unsupported": 0 },
      "Dark Mode Opacity": { "default": "", "unsupported": 0 },
      "Light Mode Opacity": { "default": "", "unsupported": 0 },
      "Disabled": { "default": 0, "unsupported": 0 },
      "Corner Radius": { "default": 0, "unsupported": 0 },
      "Enable Theme Customization": { "default": 0, "unsupported": 0 }
    },
    "Focusing": {
      "Dark Mode Gradient Color": { "default": "", "unsupported": 0 },
      "Light Mode Gradient Color": { "default": "", "unsupported": 0 },
      "Dark Mode Opacity": { "default": "", "unsupported": 0 },
      "Light Mode Opacity": { "default": "", "unsupported": 0 },
      "Disabled": { "default": 0, "unsupported": 0 },
      "Corner Radius": { "default": 0, "unsupported": 0 },
      "Enable Theme Customization": { "default": 0, "unsupported": 0 }
    },
    "Border": {
      "No Border Color": { "default": 0, "unsupported": 10 },
      "Corner Type": { "default": 0, "unsupported": 10 },
      "Dark Mode Gradient Color": { "default": "", "unsupported": 10 },
      "Light Mode Gradient Color": { "default": "", "unsupported": 10 },
      "Dark Mode Opacity": { "default": "", "unsupported": 11 },
      "Light Mode Opacity": { "default": "", "unsupported": 11 },
      "Corner Radius": { "default": 0, "unsupported": 0 },
      "Enable Theme Customization": { "default": 0, "unsupported": 0 }
    },
    "Separator": {
      "Dark Mode Gradient Color": { "default": "", "unsupported": 0 },
      "Light Mode Gradient Color": { "default": "", "unsupported": 0 },
      "Dark Mode Opacity": { "default": "", "unsupported": 0 },
      "Light Mode Opacity": { "default": "", "unsupported": 0 },
      "Disabled": { "default": 0, "unsupported": 0 },
      "Corner Radius": { "default": 0, "unsupported": 0 },
      "Enable Theme Customization": { "default": 0, "unsupported": 0 }
    }
  },
  "Tooltip": {}
}
```

Lemme know if you have any suggestions. Also, I was wondering if it would be better to put the 'DropDown' and 'Menu' tabs inside the 'General' tab as subtabs. I would really appreciate some details on the sub-tabs inside the 'Menu Items' tab.

**Reply by ALTaleX531** [reply]
Yes, as you suggested, it looks much better like this
![image](https://github.com/ALTaleX531/TranslucentFlyouts/assets/105535231/27bf9e90-2474-4871-ade8-1ec6bd04c886)

**Reply by Satanarious** [reply]
I'll make these changes ASAP.

**Comment by Satanarious** [comment]
Here's a link to my [github repo](https://github.com/Satanarious/TransparentFlyoutsConfigGUI). You can also take a look at the initial screenshots [here](https://github.com/Satanarious/TransparentFlyoutsConfigGUI/tree/master/Screenshots).

**Reply by ALTaleX531** [reply]
Wow, it look much better than I expected👍, but I have made some changes to part of the settings, and I will update the config.md later to let you know!😉

**Reply by Satanarious** [reply]
Thanks, I haven't done any styling yet. I will change the UI soon after you update the config.

**Comment by Satanarious** [comment]
I have a few doubts:
- The color and opacity parameters expect a 32-bit AARRGGBB type value. I'm using this formula to convert from RGBA, 
`AARRGGBB = (A << 24) | (R << 16) | (G << 8) | B` 
where R,G,B & A all have a value ranging from 0 to 255. Is this correct?
- The color picker doesn't let you choose the alpha value and always returns 255 as the value for A. Is the alpha value ever used for color parameters?
- If I'm not wrong then, the opacity parameters also expect a 32-bit AARRGGBB value. What are we supposed to provide as a value here, because the R,G &B dont matter here? I was wondering If I should make a simple slider from values 0 to 255 and provide the same value for R,G & B for all cases, because they don't matter here.

**Reply by ALTaleX531** [reply]
The opacity is actually a byte ranging from 0 to 255, I use (gradientColor | (opacity << 24)) to compose a 32-bit AARRGGBB type value. But for the custom rendering part, the color is BBGGRR, and I will convert it to AARRGGBB internally.

**Reply by ALTaleX531** [reply]
If you want, I can discard the Opacity key value from the next release and accept AARRGGBB using Color key value which makes the config more unified, what do you think?

**Reply by Satanarious** [reply]
The problem is, the color picker doesn't let you choose an Alpha value or I haven't researched much on it. It always returns 255 for A. My proposal is to keep the opacity value and have the color parameters accept RGB values ranging between 0 to 255 and convert both RGB and A(Opacity) all together into AARRGGBB. This way, all the logic remains in the actual program and not in the GUI
You could use the following for the registry:
- Color Parameter: `String Value` in the form of "(R,G,B)", where R,G,B range from 0 to 255
- Opacity Parameter: `Dword (32-bit) Value` in the form of integer ranging from 0 to 255

What do you think?

Edit: IGNORE THIS

**Reply by Satanarious** [reply]
Or We can go with the idea of removing the Opacity key and I find a way for the color picker to tamper the alpha value.

**Reply by Satanarious** [reply]
I found a way to tamper the alpha value, we can go with your option of removing the Opacity Key

**Reply by ALTaleX531** [reply]
OK, I will change it to complete AARRGGBB and remove the opacity key

**Reply by Satanarious** [reply]
Does the alpha value do anything in the color parameter as of now?

**Reply by ALTaleX531** [reply]
For now, the alpha value is always being ignored until the next release.

**Reply by Satanarious** [reply]
Got it. Any ETA on the next release?

**Reply by ALTaleX531** [reply]
Maybe 8.3? There is still a lot of things to test...

**Comment by Satanarious** [comment]
Let Me know when you update the config. I'm currently testing the GUI on [TF v2.0.0-alpha](https://github.com/ALTaleX531/TranslucentFlyouts/releases/tag/v2.0.0-alpha). I'm doing the registry editing bit now. It will be done in no time.

**Reply by ALTaleX531** [reply]
I was thinking how to present the config information much more elegantly since I have added a lot of configurable choices in the next update

**Reply by Satanarious** [reply]
Make a table maybe, something like this:
| **Section** | **Parameter**           | **Description**                                      | **Value Type** | **Values**                                                  | **Recommendation** | **Path**                                              |
|-------------|-------------------------|------------------------------------------------------|----------------|-------------------------------------------------------------|--------------------|-------------------------------------------------------|
| General     | `EffectType`            | Transparency effect variants on all the flyouts      | DWORD(32-Bit)  | `0`: Disable<br>`1`: Transparent<br>`2`: Solid<br>`3`: Blur | None               | `HKEY_CURRENT_USER\Software\TransparentFlyouts`       |
| General     | `EnableDropShadow`      | Choose to enable drop shadows for all flyouts or not | DWORD(32-Bit)  | `0`: False<br>`1`: True                                     | `1`                | `HKEY_CURRENT_USER\Software\TransparentFlyouts`       |
| Menu        | `DarkModeGradientColor` | Color of menu in dark more in 32-bit AARRGGBB        | DWORD(32-Bit)  | AARRGGBB                                                    | None               | `HKEY_CURRENT_USER\Software\TransparentFlyouts\Menu\` |

You can use [this](https://www.tablesgenerator.com/markdown_tables) to make tables in markdown easily

**Comment by ALTaleX531** [comment]
I have just updated the README.md and CONFIG.md (not translated yet), please have a look.

**Reply by Satanarious** [reply]
Looking good 👍

**Reply by Satanarious** [reply]
Is this final?

**Reply by Satanarious** [reply]
Also, I can do the translations, if you want

**Reply by ALTaleX531** [reply]
Then I have to trouble you. 🙏

**Reply by Satanarious** [reply]
Sure. Will do.

**Reply by Satanarious** [reply]
Sent you a pull request

**Comment by Satanarious** [comment]
You changed so much 😅. I was almost done with functionality. I'll have to redo everything. This is final right? 😅Or are more changes incoming?

**Reply by ALTaleX531** [reply]
I am sorry about that!😢 This is the final, no extra functionalities are going to be added.

**Reply by Satanarious** [reply]
Not a problem. Thanks 🙏 for making such a great piece of software. I'll go ahead and make the changes. I thought you might have kept the sections and added or removed some parameters. That is why I went on and tried to complete it. Or else, I would have waited for this release. Anyways, I'll let you know as soon as the basic functionality is complete.

**Reply by Satanarious** [reply]
Also, a doubt. Are the Color values stored in the registry as AARRGGBB or 0xAARRGGBB ?

**Reply by ALTaleX531** [reply]
It's 0xAARRGGBB.

**Comment by Satanarious** [comment]
You mentioned default values for each property in each section in the config. Does not having a key with the property name in the registry mean the default value is considered. Or is there something else going on? How do default values work in TFs?
 
 I introduced a reset button for each property where, when you left-click it, you get the default value, and when you right-click it, you get the last saved value. So consider this. The value of darkModeGradientColor is `0x412B2B2B`, if I reset to this value then this value is forced on the flyouts. Say I delete the value in the registry, what will happen then?

**Reply by ALTaleX531** [reply]
Remember the `Reading Order` in CONFIG.md? It will try to read specific keys in different registry path, if none of them exist, then it will use default value. 
So, if you delete `HKEY_CURRENT_USER\SOFTWARE\TranslucentFlyouts\DarkMode_GradientColor`, but `HKEY_LOCAL_MACHINE\SOFTWARE\TranslucentFlyouts\DarkMode_GradientColor` is still exist, then TF will not use the default value.

**Reply by Satanarious** [reply]
I understand the reading order, but your program doesn't create any registry keys, right? You have to create each on your own. So if I delete a value then its default will be used, right?

 Let's say, we start from scratch. I install Translucent Flyouts. Now what do I have in the registry? Nothing I suppose.

Also since we have nothing, then does it fall back to the default values of `HKEY_CURRENT_USER\Software\TranslucentFlyouts` or is there something else going on? Because at the beginning we don't have keys for DropDown and Menu.

Lets say I create the Menu or DropDown key and only have like 2 values in there, then what values will be used for the rest of the parameters?

My program solely works on the `HKEY_CURRENT_USER\Software\TranslucentFlyouts` path and creates keys and values there. It doesn't use any other base key apart from `HKEY_CURRENT_USER`.

I need to know what happens:
- When there is nothing in the registry for TFs
- When there are 2-3 values for TFs
- When there are 2-3 values for DropDown or Menu

By what happens I mean, what values are used and what is the fallback plan? Because I'm thinking of deleting a value or key, if a user doesn't use it or uses default values for it.

**Reply by ALTaleX531** [reply]
In fact, it is very simple to explain, if the specific key does not exist in all the paths being read, then the default value will be used. So, it will become like this.

eg. `DarkMode_GradientColor` (a common attribute, both can be used in `DropDown/Menu`)
1. `DropDown/Menu`(Found)->stop and use it.
2. `DropDown/Menu`(NotFound)->`TFs`(Found)->stop and use it.
3. `DropDown/Menu`(NotFound)->`TFs`(NotFound)->use default value.

eg. `EnableFluentAnimation` (a unique attribute)
1. `Menu`(Found)->stop and use it.
2. `Menu`(NotFound)->use default value.

Yes, when TF is installed for the first time, it will always use default values because there is nothing in the registry.
the fallback behaviour acts differently in 3 situations that you mentioned, depending on what kind of attributes they are.
So, the best way to reset is deleting the existing values.

**Reply by Satanarious** [reply]
That's a great explanation, this helps a lot, thanks.

**Comment by Satanarious** [comment]
I was testing my GUI and found something wierd.

My Registry at `HKEY_CURRENT_USER\Software\TranslucentFlyouts` looks like this, I have no sub-keys:
![image](https://github.com/ALTaleX531/TranslucentFlyouts/assets/42642682/b5a33b1b-d72d-4b13-9aae-120ffefa3bb0)
My theme accent is orange, and this is what I get on [v2.0.0-alpha.4.2](https://github.com/ALTaleX531/TranslucentFlyouts/releases/tag/v2.0.0-alpha.4.2) build:
<table>
<tr>
<td><img src="https://github.com/ALTaleX531/TranslucentFlyouts/assets/42642682/60d16e4a-4178-4c3d-bbfd-6c1382aa91e6">
<td><img src="https://github.com/ALTaleX531/TranslucentFlyouts/assets/42642682/c43cd575-a25b-42f5-9bbe-27f69be055c5" >
</table>

Since I'm almost done with functionality. Do you know what might be causing these problems, also what version do you recommend I test on? I'll release the first beta without styling as soon as the testing is done.

**Reply by ALTaleX531** [reply]
`NoBorderColor` overrides `EnableThemeColorization`, and I will fix that black popup menu issue in the v2.0.0-alpha.4.3.

**Reply by Satanarious** [reply]
So how do I achieve a menu with no borders and theme coloring ? I mean in earlier builds on enabling theme colorization the hovered over item had the theme color

**Reply by ALTaleX531** [reply]
`EnableCustomRendering` is required, and then set `EnableThemeColorization` to 1 in the `TFs\Menu\Hot`.

**Reply by Satanarious** [reply]
So what does `EnableThemeColorization` do at `TFs\` ?

**Reply by ALTaleX531** [reply]
That's for border color.

**Reply by Satanarious** [reply]
I understand now 👍. With the names, it's a little confusing 😅. I will have to mention everything in the GUI for the user.

Edit: My bad didn't read the config properly. Sorry 😅 to bother you.

**Comment by Satanarious** [comment]
Are you doing any processing with the color values which are stored as `0xAARRGGBB` ?

I have this at `TFs\` and no sub-keys:
![image](https://github.com/ALTaleX531/TranslucentFlyouts/assets/42642682/b799457f-eaf8-41fa-820f-d4a9469bd978)

This is the UI:
![image](https://github.com/ALTaleX531/TranslucentFlyouts/assets/42642682/f9d3469c-df08-4d79-acae-a67d41e0f69a)

And this is the result:
![image](https://github.com/ALTaleX531/TranslucentFlyouts/assets/42642682/52fb4928-a7ba-4985-acb7-4924cfab575b)

**Reply by ALTaleX531** [reply]
Ohh, I think I forgot to convert it, MS API uses 0xAABBGGRR internally.

**Reply by Satanarious** [reply]
Would this make you remember or should I create an issue?

**Reply by Satanarious** [reply]
Or I can do 0xAABBGGRR on my side? whichever you prefer

**Reply by ALTaleX531** [reply]
Don't worry, everything will be fixed in the next update.
I have already corrected my code and I will pushed it to github after a few hours.

**Reply by Satanarious** [reply]
Great to hear that.

**Comment by ALTaleX531** [comment]
See [v2.0.0-beta.2](https://github.com/ALTaleX531/TranslucentFlyouts/releases/tag/v2.0.0-beta.2).

**Comment by Satanarious** [comment]
Just published an alpha release for [TranslucentFlyoutsConfig](https://github.com/Satanarious/TransparentFlyoutsConfigGUI/releases/tag/v1.0.0-alpha1), would like you opinion on it.
Let me know if you'd like any changes or find any bugs.
Also, Left-click the reset button to reset the value to default and right-click to reset to previously applied value.

**Reply by ALTaleX531** [reply]
I think the GUI is almost complete except for some bugs.
1. `Enable Custom Rendering` has no effect. 
2. `Enable Fluent Animation` has no effect. 
3. `Enable Immersive Style` has no effect.

I plan to add official translucent support for tooltip in the next update but it has no additional configurable items, just like `DropDown`.
And a new functionality to remove icon background color is `Menu\ColorTreatAsTransparent` and `Menu\NoModernAppBackgroundColor`.

**Reply by Satanarious** [reply]
I'll try to fix these bugs in the meantime. And adding a few more options to Menu/DropDown will be no hassle. Thanks 😊 for taking the time to test the GUI.

Edit: I checked, I forgot to connect the values you mentioned.

**Reply by Satanarious** [reply]
I fixed everything. But I don't think I understand `DropDown` and `Focusing`. I will publish another alpha release soon. I'm using Windows 10.

Edit: Also, The explorer restarts or hangs randomly while fluent animation is enabled in the [v2.0.0-beta.2](https://github.com/ALTaleX531/TranslucentFlyouts/releases/tag/v2.0.0-beta.2)

**Comment by ALTaleX531** [comment]
However, there is still a bug. By taking a quick look to your source code (please forgive me that I don't know much about python), I think you forgot to modify `EditSettings.changeStartRatio` and some other comments.

For another thing, I wonder if it is possible to read the user's current registry value instead of reading the user's last stored value? Any interest in adding localized translations? 😊

A weird thing is, I have never seen my explorer crash or hang in my test. But I will still look into it. 🤔

**Reply by Satanarious** [reply]
Thanks, good find. I did make a mistake in the `EditSettings.changeStartRatio()` method. Will be fixed in the next alpha. About the user settings, I'm saving in a json and not fetching it directly from the registry to maintain a storage that is reliable. I usually delete keys with default values, so their values can't be fetched.

 Reading from the registry is easily possible, but I don't want someone to mash the button and make 100s of calls to the registry.

 I can maybe fetch the values from the registry on every launch and update the json based on that. What do you think?

Moreover, I'll add translation support soon.

**Reply by ALTaleX531** [reply]
It will be nice to fetch the values from the registry on every launch since every user manually edited the registry before the GUI comes out. 
By the way, I have released a new version and updated the config files.

**Reply by Satanarious** [reply]
I will add this soon. I'll have a look at the latest release and the config. Also, I'll try to add new parameters as soon as I can. I'll postpone UI styling for now and make sure every parameter is present and is flawlessly working with the import from registry feature as well.

**Comment by Satanarious** [comment]
![image](https://github.com/ALTaleX531/TranslucentFlyouts/assets/42642682/cfcbf53f-1c42-4f29-9d3e-284fcd68e0a0)
I think you missed to include possible values for this parameter.

**Reply by ALTaleX531** [reply]
It accepts a 0xAARRGGBB value just like `DarkMode_GradientColor` which also has no possible values included.

**Reply by Satanarious** [reply]
Oh! Alright.

**Comment by Satanarious** [comment]
Published a [new alpha release](https://github.com/Satanarious/TransparentFlyoutsConfig/releases/tag/v1.0.0-alpha.3) supporting the latest v2.1.0. You might wanna check it out.

**Reply by ALTaleX531** [reply]
Hmmm....🤔
![image](https://github.com/ALTaleX531/TranslucentFlyouts/assets/105535231/5f243bbb-64de-4f14-aeda-814cb6fc4967)

**Reply by Satanarious** [reply]
That's weird, I tested this on 3 devices before publishing it.
Can you tell me how to reproduce this error? Or does this not run?
 I think you have some registry values that it's not able to convert to integer. In my time of minimal testing, this didn't happen as I didn't have many values in the registry.Can you export and send me your registry values for `TFs\`.
Also, what parameter accepts a value, the error mentions as `14D`?

**Reply by ALTaleX531** [reply]
[tf.txt](https://github.com/ALTaleX531/TranslucentFlyouts/files/12470588/tf.txt)
It is apparently the issue of `PopInTime` and `FadeInTime`, this build can be successfully run after deleting these values.

**Reply by Satanarious** [reply]
I see. Thanks for the info, that helps a lot. The problem is that, when I write values to the registry they go as hexadecimals but when I read those values they return as decimals which need to be converted to hexadecimal. I convert everything except color values to integers that is what is responsible for the error. I think I should store everything as a string rather than storing any values as integers. I'll have to check how that will work. Also, thanks for the attached file.

**Comment by Satanarious** [comment]
Thank you for the earlier explanation on the bug, that helped a lot. Here is the [fixed release](https://github.com/Satanarious/TransparentFlyoutsConfig/releases/tag/v1.0.0-alpha.3) of the last version. I just want to make sure that the basic functionality is working so I can move on with additional features and styling. Thank you for testing as well. I really appreciate you help.

**Comment by Satanarious** [comment]
Also, let me know of any features you'd like on it. Here's my [tracker ](https://github.com/users/Satanarious/projects/2) just in case.

**Comment by ALTaleX531** [comment]
For now, I haven't found any bugs, well done!👍
Here are some suggestions
1. make the `General` page of `Menu` change its size according to the size of parent window
![image](https://github.com/ALTaleX531/TranslucentFlyouts/assets/105535231/5e79bf57-f9bf-44b9-8adb-aa957faed3e8)
2. add a taskbar icon like this
![TranslucentFlyouts](https://github.com/ALTaleX531/TranslucentFlyouts/assets/105535231/e61b33f1-7b10-41c7-b456-b6ba4a54d959)
3. be able to start/stop TF

I really appreciate you putting so much effort into my personal project, thanks!😊

**Reply by Satanarious** [reply]
I'll go by each suggestion you mentioned one by one:
1. That is intentional. I'm going to make the window non-resizable and remove the maximize button. I want to keep the window as small as possible.
2. I will eventually add an icon and change the current icons as placeholders. I must credit the people who made those icons and everything else. I'm leaving this part for when I do the styling, which will be right after the functionality is confirmed to work perfectly, so I can move on to the other bits that need less attention.
3. I will make sure to add options to start/stop TF along with a bunch of other features, which I mentioned in my [tracker](https://github.com/users/Satanarious/projects/2)

Thank you for making a great piece of software yourself. I initially made this for myself, but I saw many people requesting a GUI. I'm just trying to help the community so you can focus on the actual heavy lifting.

**Comment by Satanarious** [comment]
Just released my first beta. Please let me know how do you find the new changes and if you'd like anything more to be added. I would also appreciate if you could mention my repo in your readme so people can easily find my application. Thanks

**Reply by ALTaleX531** [reply]
Thanks for the long hours and effort you put into this GUI, now it really looks fantastic!👍 For me, the coolest thing is that you have embedded the configuration docs in the GUI which makes it easy to understand and convenient to use. I have already added your repo link in the readme files. Personally, I can't think of any other features that could be added, it is quite enough to be a GUI.😊

**Reply by Satanarious** [reply]
Thanks for your kind words 🙏. I will continue improving the UI further, adding more features as mentioned in my [project](https://github.com/users/Satanarious/projects/2/views/2). I'll let you know more as soon I add anything new. 
The next thing I plan to do is remove dependency on preinstalled TFs. I'd rather have the option to Download, Install, Stop, and Start TFs from within the UI.

**Comment by Satanarious** [comment]
I'm promoting our work on the windows subreddits. Might wanna check it out. Here're my posts:

-  [Windows 11](https://www.reddit.com/r/Windows11/comments/17el66e/windows32_context_menu_customization_frontend/)
-  [Windows 10](https://www.reddit.com/r/Windows10/comments/17eri56/windows32_context_menu_customization_frontend/)
-  [Windows](https://www.reddit.com/r/windows/comments/17eridl/windows32_context_menu_customization_frontend/)

**Comment by Satanarious** [comment]
Just released [v1.2.0](https://github.com/Satanarious/TransparentFlyoutsConfig/releases/tag/v1.2.0) for TFC, might wanna check it out.

**Comment by Satanarious** [comment]
I noticed as you mentioned that when I run TFC and close it, it crashes somehow. There were a few warnings, but I never used anything mentioned in those warnings, So I ignored them. I cannot find what is causing the crash even though I tried to catch the error. Some SystemExit error is caught every time I exit TFC and TF still throws an error and creates a dump.

Edit: Also, TF doesn't let me run Windbg, but as soon as I uninstall TF, it starts instantly.

**Reply by ALTaleX531** [reply]
`DisabledList` should be used in this case, using `BlockList` will result in you not being able to call exported APIs such as `StopService`. 

> Edit: Also, TF doesn't let me run Windbg, but as soon as I uninstall TF, it starts instantly.

I don't really understand what you mean, do you mean that TF is still running even after uninstalling it?

**Reply by Satanarious** [reply]
I mean, that when  I try to run Windbg it would run, but would not display I guess. But as soon as TF is uninstalled, Windbg will show up.

**Reply by ALTaleX531** [reply]
Yes, now I can confirm this. WinDBG itself didn't crash, but it just kept restarting even though it was already in DisabledList, and it wasn't until I added it to BlockList that the problem was solved. I'm not sure why this is happening, even when TF is clearly not doing anything.

**Comment by Satanarious** [comment]
So, I added TFC to DisabledList and it was still creating dumps on exit, since it was crashing. But when I added it to BlockList, it didn't load TFMain64.dll into TFC and, on exit, it didn't create any dump. Since I cannot find a way to handle the crash, should I add TFC to BlockList for the user, since context menus are intentionally disabled for my application in the code?

What do you think?

**Reply by ALTaleX531** [reply]
Maybe I'll set up another separate DisabledList for the dump settings in the next release. But until then, I'm sure this is something that can be solved with the win32 api.
![image](https://github.com/ALTaleX531/TranslucentFlyouts/assets/105535231/8b10bab6-07a3-4cda-abff-5b94986a1272)
or  
![image](https://github.com/ALTaleX531/TranslucentFlyouts/assets/105535231/750a2108-98a5-408a-b06f-9acfd12b2df1)

**Reply by Satanarious** [reply]
I'll look into this.

**Reply by Satanarious** [reply]
Well, that was quite helpful. That solves the issue completely. Thanks a ton.

**Comment by Satanarious** [comment]
Is there a need to add `/clearcache`, `/kill` & `/silent` ?

**Reply by ALTaleX531** [reply]
These commands are designed for advanced users to use and troubleshoot errors that are not intended for general use.

**Reply by Satanarious** [reply]
Because I've added `DisabledList` to each category and also adding the `BlockList` in the settings with a note asking to proceed with caution.  Or maybe I'll create a new `Advanced` tab in the settings with these settings and prompting user to proceed with caution before anything is called or triggered. What do you think?

**Reply by ALTaleX531** [reply]
I think only `BlockList` needs to be placed in the `Advanced` tab, and `DisabledList` should be available in every category for those who want to use Nilesoft Shell features as well as enjoy TF features.

**Reply by Satanarious** [reply]
Yes, you are correct. I figured that out earlier and have done the same already. What about the commands(silent, kill & clearcache) though?

Edit: I've added all the parameters in v3.0.0. Do you need anything else other than that ?

**Reply by Satanarious** [reply]
I was finishing up, I would appreciate your advice on including the above commands. Or are just `BlockList` and `DisabledList` enough?

**Reply by ALTaleX531** [reply]
`BlockList` and `DisabledList` are enough, those commands should not be added to the GUI.
